src/main/kotlin/io/github/nomisrev/repo/UserPersistence.kt

```diff
@@ -43,11 +44,14 @@ interface UserPersistence {
     bio: String?,
     image: String?
   ): Either<DomainError, UserInfo>
+
+  suspend fun unfollowProfile(followedUsername: String, followerId: UserId): Either<DomainError, Unit>
```

Comment:
This query can only fail unexpectedly, so there is no reason to use `Either`. This can also been seen because we're not using `bind`, `raise`, or any other Raise DSL method inside the implementation.

The only failure that could potentially occur is `UserNotfound`, but this is already verified and checked during authentication so we don't have to worry about it here.

```suggestion
  suspend fun unfollowProfile(followedUsername: String, followerId: UserId): Unit
```

Reply:
Committing suggestion in [9a6eba6](https://github.com/nomisRev/ktor-arrow-example/pull/190/commits/9a6eba68b857727e087a4d2930354c149f563dc9)